src/log.ts

```diff
@@ -631,12 +752,16 @@ class Log {
    * region_tag:logging_write_log_entry_advanced
    * Another example:
    */
-  write(entry, options?, callback?) {
+  write(entry: Entry|Entry[], options?: WriteOptions): Promise<ApiResponse>;
+  write(entry: Entry|Entry[], options: WriteOptions, callback: ApiResponseCallback): void;
+  write(entry: Entry|Entry[], callback: ApiResponseCallback): void;
+  write(entry: Entry|Entry[], optionsOrCallback?: WriteOptions|ApiResponseCallback, cb?: ApiResponseCallback): void|Promise<ApiResponse> {
+      const options =
+          typeof optionsOrCallback === 'object' ? optionsOrCallback : {};
+      const callback =
```

Comment:
Suggestion: is this more readable?

```ts
const [options, callback] = 
  typeof optionsOrCallback === 'function' ?
    [{}, optionsOrCallback] : 
    [optionsOrCallback, cb];
```

Reply:
That is a lot nicer :) In trying to use it though, I angered the Type gods, and ended up needing to litter more `!` around.  Since this is *such* a common pattern, do you think we should add something like this to common?

```js
  argify<T, C>(optionsOrCallback?: T, cb?: C): [T, C] {
    return typeof optionsOrCallback === 'function' ?
      [{} as T, optionsOrCallback] :
      [optionsOrCallback as T, cb];
  }
```

Reply:
💖.

Although `argify` is not a great name. How about `maybeOptionsOrCallback`?

Reply:
SGTM.  PR to implement this over in https://github.com/googleapis/nodejs-common/pull/315 :)